src/openzaak/components/autorisaties/forms.py

```diff
@@ -335,14 +357,16 @@ def get_types(self, component):
         if related_type_selection:
             _field_info = COMPONENT_TO_FIELDS_MAP[component]
 
-            # pick the entire queryset and
-            if related_type_selection in [
-                RelatedTypeSelectionMethods.all_current_and_future,
-                RelatedTypeSelectionMethods.all_current,
-            ]:
-                types = self.fields[_field_info["types_field"]].queryset
-
             # only pick a queryset of the explicitly selected objects
+            if related_type_selection == RelatedTypeSelectionMethods.select_catalogus:
+                catalogi = self.cleaned_data["catalogi"]
```

Comment:
How does this behave when no catalogus is selected?

Reply:
hmm good question, I checked it an apparently: the save continues, `self.cleaned_data["catalogi"]` is an empty QuerySet and no catalogusAutorisaties are actually created.

Come to think of it, this could also be used to "delete" existing catalogus autorisaties, so I guess this behaviour is fine? @annashamray

Reply:
yes, that's right